vng/metadata.go

```diff
@@ -294,10 +294,20 @@ func metadataValue(zctx *super.Context, b *zcode.Builder, m Metadata) super.Type
 		}
 		b.EndContainer()
 		return zctx.MustLookupTypeRecord(fields)
+	case *Named:
+		return metadataValue(zctx, b, m.Values)
 	case *Array:
+		b.Append(nil)
+		return super.TypeNull
 	case *Set:
+		b.Append(nil)
+		return super.TypeNull
 	case *Map:
+		b.Append(nil)
+		return super.TypeNull
 	case *Union:
+		b.Append(nil)
+		return super.TypeNull
```

Comment:
Nit:
```suggestion
	case *Array, *Set, *Map, *Union:
		b.Append(nil)
		return super.TypeNull
```

Reply:
Why not just put this in the default case?

Reply:
I'm going to go with default since this isn't a mission critical path and the worst thing that happens is it just doesn't prune a new type that maybe it should.